### PR TITLE
merge `new_hub` to `main`

### PR DESCRIPTION
**Pull request by reiase**
`new_hub` is a development branch for refactoring the operator/pipeline repo, in which we have finished the following features:
1.  #467 , which packages operators using the `wheel` format and manages operator dependencies for the users. The pip-based repo also resolves dependency related problems, see #513 and #536 ;
2. #523 , which supports parameterized a pipeline and allows hyper-parameter tuning;
3. #510  allows the user to declare an operator in jupyter notebook with a single code block, which makes operator development much easier;
4. #526 , which supports running an operator directly and makes operator debugging much easier;
5. #485 , which allows the user to run a pipeline from cmdline interface;

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
 pyyaml>=5.3.0
 requests>=2.12.5
-tqdm>=4.59.0
\ No newline at end of file
+tqdm>=4.59.0
+gitpython>=3.1
```
Comment (shiyu22) [inline requirements.txt:4]: No need for `gitpython` now, use subprocess with `git` command instead in [git_utils.py](https://github.com/towhee-io/towhee/blob/main/towhee/utils/git_utils.py).

**Review by GuoRentong — CHANGES_REQUESTED** [review]
I suggest to flatten all the commands and add the necessary ones:
* create-op
Create an empty operator based on template
* create-pipeline
Create an empty pipeline based on template
* add
Add a local op/pipeline to workspace
* install
Install an op/pipeline
* clone
Clone op/pipeline repo
* package
Package op/pipeline in wheel format
* push
Push op/pipeline repo to hub
* execute
Execute a pipeline

in addition, `develop --develop` is somewhat misleading, names like `add` maybe better

File: requirements.txt (hunk 1 above)
Comment (reiase) [reply requirements.txt]: fixed

**Comment by reiase** [comment]
> I suggest to flatten all the commands and add the necessary ones:
> 
> * create-op
>   Create an empty operator based on template
> * create-pipeline
>   Create an empty pipeline based on template
> * add
>   Add a local op/pipeline to workspace
> * install
>   Install an op/pipeline
> * clone
>   Clone op/pipeline repo
> * package
>   Package op/pipeline in wheel format
> * push
>   Push op/pipeline repo to hub
> * execute
>   Execute a pipeline
> 
> in addition, `develop --develop` is somewhat misleading, names like `add` maybe better

will fix these problems in #561

**Pull request merged**